spec/controllers/blackouts_controller_spec.rb

```diff
@@ -116,53 +116,112 @@
         end
       end
     end
+
     context 'POST create' do
-      context 'with correct params' do
-        before do
-          @attributes = FactoryGirl.attributes_for(:blackout)
-          post :create, blackout: @attributes
-        end
+      shared_examples_for 'creates blackout' do |attributes|
+        before { post :create, blackout: attributes }
+
         it 'should create the new blackout' do
           expect(Blackout.find(assigns(:blackout).id)).not_to be_nil
         end
         it 'should pass the correct params' do
-          expect(assigns(:blackout)[:notice]).to eq(@attributes[:notice])
+          expect(assigns(:blackout)[:notice]).to eq(attributes[:notice])
           expect(assigns(:blackout)[:start_date]).to\
-            eq(@attributes[:start_date])
-          expect(assigns(:blackout)[:end_date]).to eq(@attributes[:end_date])
+            eq(attributes[:start_date])
+          expect(assigns(:blackout)[:end_date]).to eq(attributes[:end_date])
           expect(assigns(:blackout)[:blackout_type]).to\
-            eq(@attributes[:blackout_type])
+            eq(attributes[:blackout_type])
         end
         it { is_expected.to redirect_to(blackout_path(assigns(:blackout))) }
         it { is_expected.to set_flash }
       end
-      context 'with incorrect params' do
+
+      shared_examples_for 'does not create blackout' do |attributes|
```

Comment:
Scratch that, we refer to the `attributes` variable in some of our `it` blocks so they need to be within the entire shared example context, hence passed in. Given that this would be a minor difference I'm just going to merge it as-is, if that's ok with you.